src/core/hex.ts

```diff
@@ -131,8 +131,11 @@ export class Hex {
    * @returns Hex
    */
   static fromHexInput(hexInput: HexInput): Hex {
-    if (hexInput instanceof Uint8Array) return new Hex(hexInput);
-    return Hex.fromString(hexInput);
+    // Note we have to be more defensive here, since Javascript doesn't have the concept of types
+    if (typeof hexInput === "string") {
```

Comment:
not completely sure this is a fix. HexInput is of type string | Uint8Array. Hex constructor accepts Uint8Array. if hexInput is of type string, it would never be instanceof Uint8Array and therefore would get to Hex.fromString. 

The thing is that javascript doesnt care for types, so technically you could pass in whatever you want (tho I think you would get a typing error from the sdk). So if you pass in anything that is not string | Uint8Array you.... screwed.